### PR TITLE
Refactor History Sync

### PR DESCRIPTION
**Pull request by amir-blox**

### CODE REVIEW

File: ibft/sync/history.go [hunk 1]
```diff
@@ -138,54 +111,90 @@ func (s *HistorySync) findHighestInstance() (*proto.SignedMessage, string, error
 			continue
```
Comment (alonmuroch) [inline ibft/sync/history.go:111]: @amir-blox isn't this redundant?

File: ibft/sync/history.go [hunk 2]
```diff
@@ -100,36 +100,9 @@ func (s *HistorySync) findHighestInstance() (*proto.SignedMessage, string, error
 		usedPeers = usedPeers[:4]
 	}
 
-	// fetch response
```
Comment (alonmuroch) [inline ibft/sync/history.go:103]: @amir-blox maybe we should expand the function description to also include expected behaviour

File: ibft/sync/incoming.go [hunk 3]
```diff
@@ -81,22 +81,23 @@ func (s *ReqHandler) handleGetDecidedReq(msg *network.SyncChanObj) {
 // handleGetHighestReq will return the highest known decided msg.
 // In case there isn't one, it will return a 0 length array
 func (s *ReqHandler) handleGetHighestReq(msg *network.SyncChanObj) {
+	res := &network.SyncMessage{
+		Lambda:         s.identifier,
```
Comment (alonmuroch) [inline ibft/sync/incoming.go:85]: @amir-blox we should add tests for this

File: ibft/sync/history_test.go [hunk 4]
```diff
@@ -281,63 +281,6 @@ func TestFindHighest(t *testing.T) {
 			10,
 			"",
 		},
-		//
```
Comment (alonmuroch) [inline ibft/sync/history_test.go:284]: @amir-blox did we cover all scenarios in the tests?

**Review by alonmuroch — CHANGES_REQUESTED** [review]

File: ibft/sync/history_test.go (hunk 4 above)
Comment (amir-blox) [reply ibft/sync/history_test.go:284]: The following scenarios are covered:
1. valid
2. empty responses
3. error responses
4. some errors, some valid
5. valid multi responses
6. valid multi responses different seq

I've added another scenario - "some errors, some empty"
Cn you think of other scenarios besides that?

Comment (amir-blox) [reply ibft/sync/history_test.go:284]: Added scenarios:
* all invalid
* some invalid, some empty

**Review by alonmuroch — APPROVED** [review]

**Pull request merged**